src/tests/bonus.test.js

```diff
@@ -4,74 +4,73 @@ import { fireEvent, screen, waitFor } from '@testing-library/react';
-      renderWithRouterAndStore(<Wallet />, '/carteira', initial);
-      expect(screen.getAllByTestId('edit-btn')[0]).toBeInTheDocument();
-    });
-  
-    test('Ao ser clicado, o botão habilita um formulário para editar a linha da tabela. Ao clicar em "Editar despesa" ela é atualizada e atualiza a soma de despesas no header.', async () => {
-      const { store } = renderWithRouterAndStore(<Wallet />, '/carteira', initial);
-      const toggleEditBtn = screen.getAllByTestId('edit-btn')[0];
-      fireEvent.click(toggleEditBtn);
-  
-      const valueInput = await screen.findByTestId('value-input');
-      const currencyInput = await screen.findByTestId('currency-input');
-      const methodInput = await screen.findByTestId('method-input');
-      const tagInput = await screen.findByTestId('tag-input');
-      const descriptionInput = await screen.findByTestId('description-input');
-      const editButton = await screen.findByText(/Editar despesa/i);
-  
-      userEvent.type(valueInput, '100');
-      userEvent.selectOptions(currencyInput, 'CAD');
-      userEvent.selectOptions(methodInput, 'Dinheiro');
-      userEvent.selectOptions(tagInput, 'Trabalho');
-      userEvent.type(descriptionInput, 'Cem dólares canadenses');
-  
-      fireEvent.click(editButton);
-  
-      await waitFor(() => {
-        expect(
-          screen.getByRole('cell', { name: 'Cem dólares canadenses' })
-        ).toBeInTheDocument();
-      });
-  
-      expect(screen.getAllByRole('cell', { name: 'Trabalho' })[0]).toBeInTheDocument();
-      expect(screen.getAllByRole('cell', { name: 'Dinheiro' })[0]).toBeInTheDocument();
-      expect(screen.getAllByRole('cell', { name: '100' })[0]).toBeInTheDocument();
-      expect(screen.getAllByRole('cell', { name: 'Dólar Canadense' })[0]).toBeInTheDocument();
-      expect(screen.getAllByRole('cell', { name: '4.20' })[0]).toBeInTheDocument();
-      expect(screen.getAllByRole('cell', { name: '420.41' })[0]).toBeInTheDocument();
-      expect(screen.getAllByRole('cell', { name: 'Real' })[0]).toBeInTheDocument();
-  
-      const newExpenses = [
-        {
-          id: 0,
-          value: '100',
-          currency: 'CAD',
-          method: 'Dinheiro',
-          tag: 'Trabalho',
-          description: 'Cem dólares canadenses',
-          exchangeRates: mockData,
-        },
-        {
-          id: 1,
-          value: '20',
-          currency: 'EUR',
-          method: 'Dinheiro',
-          tag: 'Trabalho',
-          description: 'Vinte euros',
-          exchangeRates: mockData,
-        },
-      ];
-  
-      expect(store.getState().wallet.expenses).toStrictEqual(newExpenses);
+describe('11 - Crie um botão para editar uma despesa da tabela contendo as seguintes características:', () => {
+  const initial = initialStateWithExpenses;
+
+  test('O botão deve estar dentro do último item da linha da tabela e deve possuir `data-testid="edit-btn"`', () => {
```

Comment:
Não está sendo testado se o botão se encontra na última posição da tabela, então acho que pode ter um testo mais genérico
```suggestion
  test('O botão deve estar dentro da linha da tabela e deve possuir `data-testid="edit-btn"`', () => {
```